---
title: "Understanding the technological transition based on cultural transmission biases during the Late Pleistocene in Korea"
author:
  - Gayoung Park
  - Ben Marwick
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
    bookdown::html_document2:
      fig_caption: yes
      reference_docx: "../templates/template.docx" # Insert path for the DOCX file
bibliography: references.bib
csl: "../templates/journal-of-archaeological-science.csl" # Insert path for the bib-style
abstract: |
  Text of abstract
keywords: |
  keyword 1; keyword 2; keyword 3
highlights: |
  These are the highlights. 
---


<!-- This is the format for text comments that will be ignored during renderings. Do not put R code in these comments because it will not be ignored. -->

```{r, setup, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  echo = FALSE,
  comment = "#>",
  fig.path = "../figures/"
)

#library(CTtps) # Or use devtools::load_all('.', quiet = T) if your code is in script files, rather than as functions in the `/R` diretory

library(here)
library(tidyverse)
```

# Introduction

1. What are our research questions?
- Understanding technological transition during the Late Pleistocene in Korea base on Cultural transmission. 
- What are the social contexts led to the technological transition

2. What is our archaeological background?
-SP was started in Korean Peninsula -> spread to Japan
-Why SP important, Why we care
-Current explantion of SP dispersal: migration vs in situ, in between
 The evidence of the two main models: proportion of SP in assemblage, Y chromosome (two groups of people), composition of assemblage (blade to flake ratio)
 -> no explanation about any mechanism/process/driver for the new technology
 -> lack of details of the lithic technology, human behavior

3. What is our conceptual framework?
- Evolutionary concepts that links stone artifacts to human behavior
- Cultural Evolution : the idea that cultural change constitutes an evolutionary process
- Same general concept of biological evolution: genetic change in population that is inherited over generations. Apply this concept to artifacts or cultural phenomena

- In this paper, I will use Cultural Transmission 
- Cultural Transmssion: way that people learn and pass on information
- Transmission biases
- similar study (CT in stone artifacts) : Mesoudi&O'Brien, Bettinger&Eerkens, Garvey

-------------
Application the theoretical and methodological tenets of Darwinian evolutionary concpet to archaeological studies has helped understand technological transition related to human behavior (Dunnell 1980; Lipo et al. 1997; Bettinger and Eerkens 1997, 1999, 2008, Mesoudi abd O'Brien 2008). 

these approaches build on earlier attempts to model cultural change according to evolutionary principles. ). In the case of culture, the mechanism
 of inheritance is the process of social transmission, which is comprised of several different potential learning mechanisms, such as stimulus enhancement, emulation, imitation, and teaching (O'Brien and Lyman 2000, Schillinger et al. 2014)

need to add 'biological concept to archaeological usage': (Richerson and Boyd 1992)


<<<<<<< HEAD

Cultural evolution that archaeologists adapted has three sub-categories: human behavioral ecology, phylogenetics and cultural transmission theory (Garvey 2018). Among them, we apply models and methods of cultural transmission to understand the technological change and socio cultural contexts. 
=======
Cultrual evolution that archaeologists adapted has three sub-categories: human behavioral ecology, phylogenetics and cultural transmission theory (Garvey 2018). Among them, we apply models and methods of cultural tranmssion to understand the technological change and socio cultural contexts. 
>>>>>>> b0e928d049d619a5fc6902ad1ff878778d5bd355



Cultural transmission is the idea that a transition can occur during the process of social learning and the transmission biases are important loci of changes in material cultural (Lycett 2015; Creanza et al. 2017). In other words, this concept holds that instructions for making artifacts are transmitted between members of a group during social learning by observation of, or interaction with another member (Heyes 1994; Eerkens and Lipo 2007; Brantingham et al. 2010; Lycett 2015). 

Variation in the raw material on which selection operates to cause changes in the frequency of cultural traits through time (Eerkens and Lipo 2005)

 Copying errors that occur during the manufacture of artifactual traditions are potentiall (Schillinger et al. 2014)

The transmission is subject to biases that result from different social contexts. Boyd and Richerson (1985) and Henrich and McElreath (2003) have defined many of these biases, including ‘guided variation’ (where individuals learn new behaviors and then modify them through trial and error), ‘content-based bias’ (where some aspect of the transmitted instructions, such as cultural preferences, makes them more likely to be adopted), ‘frequency-based bias’ (where an individual is biased to choose particular instructions based on their perceived frequency in the population, such as extremely popular or rare instructions), and ‘model-based bias’ (where a variant is transmitted because of its association with other attributes, such as the prestige or skill of other individuals). <!-- add some description of experimental and ethnographic evidence that these biases are real, some of this not archaeological, e.g. https://www.pnas.org/content/116/14/6726  --> 
For example, the influence of transmission biases on change and variation, such
as preferentially copying prestigious or successful individuals (indirect bias), has been demonstrated using explicit archaeological examples such as Great Basin projectile points (Bettinger and Eerkens 1999; Mesoudi and O'Brien 2008, Schillinger et al. 2014)
 
Bettinger and Eerkens (1997; 1999; 2008) have shown how these biases can be identified in stone artifact assemblages. They studied metric variables of stone points during the introduction of bow and arrow technology in the Great Basin around AD 300-600. They equated guided transmission with high metric variation and low correlation between metric attributes, and equated model-based bias with less variation and more correlated variables. They found that in eastern California artifacts have poor correlations of basal width and mass, suggesting new technology was introduced and spread by guided transmission.On the other hand, in central Nevada metric variables are highly correlated, indicating that the bow and arrow was introduced and spread by model-based bias. Since Bettiniger and Eerkens studied archaeological data so they are unable to directly observe the transmission process. 
Eerkens and Bettinger (2001) claims a robust statistical technique to compare varations within different kinds of assemblages and propose actual values of the coefficient of variation for showing whether artifacts are randomly produced (CV = 57.7 percent) or the result of planned production (CV 1.7 percent). 

To test the underlying assumptions that Bettinger and Eerkens used, Mesoudi and O’Brien (2008) have conducted experiments that simulated prestige bias and guided variation, and observed results that supported Bettinger and Eerkens’ hypothesis.

experimental studies related to cultural transmission
- large populations generate greater artifact variations but do not enhance artifact performance (Fay et al. 2019)
- material items are produced via "additive" processes such as pottery and produced via "reductive" processes like stone artifact. Statistically 
(Schillinger et al. 2014)


Garvey (2018)
-------------


4. What are our methods?
- Framework: Bettinger&Eerkens

To investigate the specific relationship between technological innovations in Korea and the influence of social contexts, I will apply Bettinger and Eerkens’ (1999) approach because their focus on the introduction of new tool in a region is closely related to my research question about new technologies in Late Pleistocene Korea. I will focus on stemmed points in the Korean Paleolithic, a distinctive kind of projectile point similar to the points studied by Bettinger and Eerkens, to study transmission biases. I propose two models to investigate the social context of new technologies in Korea: a single-point origin model and multi-points origins model. 

The single-point origin model proposes that the new technology appeared at a single point in Korea, and spread out from this single origin location via relatively long-distance trips, or a chain of direct transmission through multiple individuals. This model is similar to Bae’s migration model for the introduction of new tools in the Korean Paleolithic, but is not identical because the single point in my model could be a single point of indigenous innovation, rather than an external introduction (distinguishing between these two possibilities is beyond the scope of my project). This model predicts that if the new technologies appeared from a single origin, then model-based bias will have been important in the spread of the technology, and the metric attributes of artifacts from my sites will be strongly correlated, and there will be low variation between sites. 

The multiple-points origin model proposes that new technology appeared independently at multiple locations on the Korean peninsula. This is analogous to Seong’s ‘in situ’ model of local innovation. This model proposes that the social context for the appearance of new tools is one of isolated groups with limited contact, where guided variation dominates transmission biases and innovations in material culture that are strongly determined by trial and error. This model predicts that attributes of artifacts will be poorly correlated and show high variation between sites.

- Two models: Guided variation and Model-based bias =>  Multi-point Origin (in situ or in between one) and Single-point Origin (migration model)
- Test by Correlation and Variation (Pearson's r, PCA, Coefficient of Variation) 
- data collection, sites, artifact #
- attributes: landmarks (tps software)

5. What are our results? 
- overall change through time
- correlation: postively correlated(?). We are using PCA too
- PCA: 2,3 phases are highly overlapped, high redundancy. The individual phase are scattered, low redundancy. Variables of Dimension 1 is positively correlated
- CV:values are around 20(ML)-34(SL). Compared to 2nd phase, 3rd phase has slightly lower variation.   

6. What is the answer to our questions??
- low correlation + high variation? High correlation + low variation?
- What are the criteria for correlation and variation
- How was SP spread in Korea, did they have single orgin or multiple origins?
- What are the impact of combinaton with blade technology

7. What is limitation of this study?
- Maybe applying CV for SP is not ideal. It can be depending on individual knapper's skill and raw materials
- CV of one site (SYG6) - no big difference with the value of all data
- CV: lithis vs pottery

- couldn't include thickness and weight

** Expectation
- The main measurements are correlation(+PCA) and CV.
- The set of the two measurements would be like: 
1) lower CV and higher correlation between sites, in this case, we will think that SPs were the result of model based bias and possibly had single origin. 
2) higher CV and lower correlation between sites, for this case, we will thank that SPs were the result of guided variation and possibly had multiple origins. 
- We have three phases, but there are only two artifacts for the first phase. There are more artifacts and sites for the first phase but not eligible for this research because we gathered complete or almost complete shape of SPs. We excluded broken pieces. Therefore, we focus on the change from the 2nd phase to 3rd phase for understanding the process of SP's dispersal in the Korean Peninsula.
- Change from the 2nd to 3rd
1) less correlated: the two groups were not very related/locally developed 
2) highly correlated: the two groups were related, had single source and then spread to other places
- Site by site
- Regional relationship? Province/cities


# Background


# Methods

Reading tps data and phase information of each artifact 

```{r results='hide', include=FALSE}
#reading landmarks in tps, code from https://gist.github.com/mrdwab/2062329


read.tps = function(data) {
  # Reads the .tps file format produced by TPSDIG 
  # (http://life.bio.sunysb.edu/morph/ into a single data frame
  # USAGE: R> read.tps("filename.tps")
  a = readLines(data) # so we can do some searching and indexing
  LM = grep("LM", a) # find the line numbers for LM
  ID.ind = grep("ID", a) # find the line numbers for ID
  # and the ID values, SCALE values, and image names
  ID = gsub("(ID=)(.*)", "\\2", grep("ID", a, value=T)) 
  SCALE = gsub("(SCALE=)(.*)", "\\2", grep("SCALE", a, value=T)) 
  images = basename(gsub("(IMAGE=)(.*)", "\\2", a[ID.ind - 1]))
  # FOR EACH LOOP   
  skip = LM # set how many lines to skip
  # and how many rows to read
  nrows = as.numeric(gsub("(LM=)(.*)", "\\2", grep("LM", a, value=T)))
  l = length(LM) # number of loops we want
  
  landmarks = vector("list", l) # create an empty list
  
  for (i in 1:l) {
    landmarks[i] = list(data.frame(
      read.table(file = data, header = F, skip = LM[i],
                 nrows = nrows[i], 
                 col.names = c("X", "Y"),
                 stringsAsFactors = TRUE),
      IMAGE = as.character(images[i]),
      ID = ID[i],
      SCALE = SCALE[i],
      stringsAsFactors = TRUE))
  }
  do.call(rbind, landmarks) # rbind the list items into a data.frame
}

#need to figure out how to read the file inside the project
landmarks <- read.tps(here("analysis/data/raw_data/BG_0_2.tps"))

#getting info for phase of each site
SPstage <- read_csv(here("analysis/data/SPstage.csv"))

#Compare multiple artifacts 
#Open multiple tps files. This will need to be changed to read files inside the project
datadir <- here("/analysis/data/raw_data/")
files <- dir(datadir, pattern = "*.tps$|*.TPS$")
```


Computing size attributes  of the points by calculating distance between landmarks
```{r}

#Understanding each landmark
#ML: maximum length, tip to bottom, perpendicular to the length axis 1- 7
#BL: body length: tip to the closest wing, perpendicular to the length axis 1- 3 OR 1-10 depending on the artifact
#TL: Tang length, the closest wing from the tip to bottom, perpendicular to the length axis 9 - 7 depending on the artifact
#SL: max stem length, perpendicular to the length axis. Closer tang curve’s middle point from the tip to the most distant point of the basal end  9-7 depending on the artifact
#MW: mid width: dimension from margin to margin at the mid-point of the length 2 - 11 axis, perpendicular to the width axis.
#TW: tang width: dimension between each wing, perpendicular to the width  axis  3 - 10
#SW: stem width: width of the basal end of the point, 5mm above the end  5 - 8

#distance between landmarks(euclidean)

landmark_dist = function(landmarks, pt1, pt2) {
  sqrt(
    (landmarks$X[pt1] - landmarks$X[pt2])^2 + 
      (landmarks$Y[pt1] - landmarks$Y[pt2])^2
  ) * as.numeric(levels(landmarks$SCALE))
}

# distance between two y axis 
vertical_dist = function(landmarks, pt1, pt2) {
  d = abs(landmarks$Y[pt1] - landmarks$Y[pt2])
  d * as.numeric(levels(landmarks$SCALE))
}


#Maximum length of sp: ML, distance between landmark 1-7
ML <- landmark_dist(landmarks, 1, 7)

#Body length of sp is 1-3 or 1-10, depending on each artifact.
bl_dist = function(landmarks) {
  d1 = landmark_dist(landmarks, 1, 3)
  d2 = landmark_dist(landmarks, 1, 10)
  max(d1, d2)
}

BL <- bl_dist(landmarks)

# Tang length of sp: TL, distance between landmark 10-7 or 3-7, (y axis)
tl_dist <- function(landmarks) {
  t1 = vertical_dist(landmarks, 10, 7)
  t2 = vertical_dist(landmarks, 3, 7)
  max(t1, t2)
}

TL <- tl_dist

# Maximum length of stem: SL, distance between landmark 9-7 or 4-7 (y axis)
sl_dist <- function(landmarks) {
  s1 = vertical_dist(landmarks, 9, 7)
  s2 = vertical_dist(landmarks, 4, 7)
  max(s1, s2)
}

SL <- sl_dist

# Mid width: MW, distance between landmark 2-11
MW <- landmark_dist(landmarks, 2, 11)

# Maximum width of Tang: TW, distance between landmark 3-10
TW <- landmark_dist(landmarks, 3, 10)

# Stem width: SW, distance between landmark 5-8
SW <- landmark_dist(landmarks, 5, 8)

```

Gather attributes from  all artifacts

```{r}

ML = c()
BL = c()
TL = c()
SL = c()
MW = c()
TW = c()
SW = c()
for (i in 1:length(files)) {
  filename <- paste(datadir, files[i], sep = "")
 # print(filename)
  landmarks <- read.tps(filename)
  ML <- c(ML, landmark_dist(landmarks, 1, 7))
  BL <- c(BL, bl_dist(landmarks))
  TL <- c(TL, tl_dist(landmarks))
  SL <- c(SL, sl_dist(landmarks))
  MW <- c(MW, landmark_dist(landmarks, 2, 11))
  TW <- c(TW, landmark_dist(landmarks, 3, 10))
  SW <- c(SW, landmark_dist(landmarks, 5, 8))
}

#list of all attributes
df <- data.frame(ML, BL, TL, SL, MW, TW, SW, files, SPstage$Stage)

#trim the name of each artifact.
df_sitename <- df %>% 
  mutate (files = str_remove_all(files, "_.tps")) %>% 
  separate (files, into = c("sitename", "files"), sep = "_") %>% 
  select (-files)

#list of all attributes without ID
DF <- df[,1:7]
```

Box plot for each attributes
```{r}
box1 <- boxplot(DF, main = "Variation for each attributes", col = "gold")
```

This boxplot indicates the overview of each attribute. We can see that ML and BL have wider variation compared to other attribtues. Since TL has relativelly less variable, ML variation is more depending on BL. Throughout time and region, dimension of SW is consistant.   

Attributes that we are using here

 It must be noted that all of this previously undertaken work has focused exclusively on the size (i.e., scale) of artifactual attributes and the patterns of variation that may be produced via copying error rates due to limits in size perception. However, variation in the overall shape of artifacts is also evidently important, given that aspects of shape may have specific functional or aesthetic properties (Schillinger et al. 2014). 



Computing Coefficient or variation
```{r}
# Coefficient of variation (CV)

cv <- function(x, ... ) sd(x, ...) / mean(x, ...) *100

# we've got DF a data frame with cols as variables
# rows and specimens, and values as dimensions
# need to add a column of site ID
cvmap <- map(DF, cv)

#Same as map but better view for summary
map_df(DF, cv)


```

The coefficient of variation (CV) is the ratio of the standard deviation devided by mean. It shows the extent of variability from the mean. 

CV for all attributes are distributed from 23-31%. Interestingly, SL and SW have high CV values (36.3 and 32.0, respectively). These attributes show lower variation on boxplot. <= need to think about

Make a table of CVs for all variable grouped by site

```{r}
cv_by_site_df  <- 
df_sitename %>% 
  select(-SPstage.Stage) %>% 
  group_by(sitename) %>% 
  add_tally() %>% 
  filter( n > 1) %>% 
  select(-n) %>% 
 nest(-sitename) %>% 
  mutate(cv_by_site = map(data, ~map_df(.x, cv)))
  
cv_by_site_df_unnest <-   
cv_by_site_df %>% 
  unnest(cv_by_site)


cv_plot_site <- cv_by_site_df_unnest %>% 
  select(-data) 

library(tidyr)
cv_plot_site %>% 
pivot_longer(cols = -sitename, names_to = "group") %>% 
ggplot(aes(x = group, y = value, fill = as.factor(sitename))) + 
geom_bar(stat="identity", position=position_dodge())

```

The plot shows CV values for different attributes The color of the bars represent different sites. Among tehs sites, we can see SC (Sachang) site has significantly high CV values. SC has two artifacts, one is from 1st phase and the other one is 2nd phase. We assume that this time difference and few number of artifacts may relate to the high CV value.   

Unlike the previous plot, in this one, the bars represent different attributes. You can see that SC has highest CV values of most attributes. Overall, SW values are very high. 

Make a table of CVs for all variable grouped by phase

BM: discuss this one first, this is the overview picture for our story:  

```{r}
cv_by_stage_df  <- 
df_sitename %>% 
  select(-sitename) %>% 
  group_by(SPstage.Stage) %>% 
  add_tally() %>% 
  filter(SPstage.Stage > 1) %>% 
  nest(-SPstage.Stage) %>% 
  mutate(cv_by_stage = map(data, ~map_df(.x, cv)))
  
cv_by_stage_df_unnest <-   
cv_by_stage_df %>% 
  unnest(cv_by_stage)

cv_plot_stage <- cv_by_stage_df_unnest %>% 
  select(-data, -n)

library(tidyr)
cv_plot_stage %>% 
pivot_longer(cols = -SPstage.Stage, names_to = "group") %>%
ggplot(aes(x = group, y = value, fill = as.factor(SPstage.Stage))) + 
geom_bar(stat="identity", position=position_dodge()) + labs(x="Attributes", fill= "Phase")
```

This plot shows the difference of CV in between 2nd and 3rd phase. For TL, both two phases have almost same values. For MW and TW, 3rd phase has lower values than the 2nd. The other 4 attributes show higher valeues in 3rd phase, which means less standardized in their shape during the latter period.

By the way, Previous research (Bettinger&Eerkins, Garvey) focused on Width of the projectile points along with weight to understand cultural transmssion. We haven't figured out which attibutes would be more infortant than others yet. 

BM: this discuss this one second, this is where we get into the details 

```{r}
## CV with site (n=X)


 cv_by_site_df_label <- 
df_sitename %>% 
  select(-SPstage.Stage) %>% 
  group_by(sitename) %>% 
  add_tally() %>% 
  filter( n > 1) %>% 
   mutate(label = as.factor(paste0(sitename,' (N = ', n, ")"))) %>%   select(-n) %>% 
 nest(-sitename, -label) %>% 
  mutate(cv_by_site = map(data, ~map_df(.x, cv)))

cv_by_site_df_unnest_label <-   
cv_by_site_df_label %>% 
  unnest(cv_by_site)

cv_plot_site_label <- cv_by_site_df_unnest_label %>%
                select(-data)





## Reference: site and number of artifacts
 cv_by_site_df_label_1 <- 
df_sitename %>% 
  select(-SPstage.Stage) %>% 
  group_by(sitename) %>% 
  add_tally() %>% 
   mutate(label = as.factor(paste0(sitename,'\nN = ',n))) %>%   select(-n) %>% 
 nest(-sitename, -label) %>% 
  mutate(cv_by_site = map(data, ~map_df(.x, cv)))

cv_by_site_df_unnest_label_1 <-   
cv_by_site_df_label_1 %>% 
  unnest(cv_by_site)

library(tidyr)
p<- cv_plot_site_label %>% 
pivot_longer(cols = -c(sitename,label), names_to = "group") %>% 
ggplot(aes(label, value, fill = as.factor(group))) + 
geom_bar(stat="identity", position=position_dodge())

p + labs(x= "Site", fill = "Attributes")
```

Unlike the previous plot, in this one, the bars represent different attributes. You can see that SC has highest CV values of most attributes. Overall, SW values are very high. 

```{r}
# plot CVs by site with number of artefacts showing >10
# phase 2 and phase 3 sites only
cv_by_site_df_label_more_than_10 <- 
df_sitename %>% 
  unite(col = "site_phase", 
        c(sitename, 
          SPstage.Stage), 
        sep = "_") %>% 
  group_by(site_phase) %>% 
  add_tally() %>% 
  filter( n > 5) %>% 
   mutate(label = as.factor(paste0(site_phase,' (N = ', n, ")"))) %>%   select(-n) %>% 
 nest(-site_phase, -label) %>% 
  mutate(cv_by_site = map(data, ~map_df(.x, cv))) %>% 
  unnest(cv_by_site) %>% 
  select(-data)

cv_by_site_df_label_more_than_10 %>%
  pivot_longer(cols = -c(site_phase, label), 
               names_to = "lithic_attribute",
               values_to = "CV") %>%
  ggplot(aes(x = lithic_attribute, 
             y = CV)) +
  geom_col() +
  facet_wrap(~  label) +
  ggtitle("CVs of point attributes by site (with sites >10 artefacts")

cv_by_site_df_label_more_than_10_ml_mw_tw <- 
cv_by_site_df_label_more_than_10 %>%
  pivot_longer(cols = -c(site_phase, label), 
               names_to = "lithic_attribute",
               values_to = "CV") %>%
  filter(lithic_attribute %in% c("ML", "MW", "TW"))  %>% 
  mutate(phase = parse_number(str_extract(site_phase, "_."))) %>% 
  group_by(site_phase) %>% 
  mutate(mean_cv = mean(CV))
  
  ggplot(cv_by_site_df_label_more_than_10_ml_mw_tw,
         aes(x = lithic_attribute, 
             y = CV)) +
  geom_col() +
  geom_hline(aes(yintercept = mean_cv), 
             colour = "red") +
  facet_wrap(~  label) +
  ggtitle("CVs of point attributes by site (with sites >10 artefacts)") +
  theme_minimal()
```

```{r}

cv_by_site_df_label_more_than_10_ml_mw_bl <- 
cv_by_site_df_label_more_than_10 %>%
  pivot_longer(cols = -c(site_phase, label), 
               names_to = "lithic_attribute",
               values_to = "CV") %>%
  filter(lithic_attribute %in% c("ML", "BL","MW"))  %>% 
  mutate(phase = parse_number(str_extract(site_phase, "_."))) %>% 
  group_by(site_phase) %>% 
  mutate(mean_cv = mean(CV))
  
  ggplot(cv_by_site_df_label_more_than_10_ml_mw_bl,
         aes(x = lithic_attribute, 
             y = CV)) +
  geom_col() +
  geom_hline(aes(yintercept = mean_cv), 
             colour = "red") +
  facet_wrap(~  label) +
  ggtitle("CVs of point attributes by site (with sites >=7 artefacts)") +
  theme_minimal()
```

```{r}
#using corrr package
library(tidyverse)
library(corrr)

corrr_all <- correlate(DF) %>% 
  shave()

call<-rplot(corrr_all) +
  ggtitle("Correlation for all phase") 

direction_all <- correlate(DF) %>% 
  network_plot()

#phase 2

DF2<-df_sitename %>% 
  filter(SPstage.Stage == 2) %>% 
  select(-sitename, -SPstage.Stage)

corrr_2 <-correlate(DF2) %>% 
  shave()

ctwo <- rplot(corrr_2) +
  ggtitle("Correlation for 2nd phase") 

direction_2 <- correlate(DF2) %>% 
  network_plot()

#phase 3

DF3<-df_sitename %>% 
  filter(SPstage.Stage == 3) %>% 
  select(-sitename, -SPstage.Stage)

corrr_3 <-correlate(DF3) %>% 
  shave()

cthree <- rplot(corrr_3) +
  ggtitle("Correlation for 3rd phase") 

direction_3 <-correlate(DF3) %>% 
  network_plot()
  


### three phases at once
library(patchwork)
call + ctwo + cthree + plot_layout(ncol=1)

```

When it comes to the correlations between seven attributes, BL-ML, TW-MW, TL-TW, and TL-SL are relatively close to each other. There is no negative correlation among the attributes. Compared to the 2nd phase, in 3rd phase, correlations of SL with TW and MW, and TL with TWm SL, and MW get stronger. On the other hand, the correlation of TW-SW becomes weeker. Overall, the correlation between attributes became stronger from 2nd to 3rd phase. 

Combined the two results of CV and correlation from 2nd to 3rd... 

CV: lower in MW and TW, but higher in other 5 attribtues.

COR: overall correlation became stronger. 

=> This combination doesn't fit into the models.


Eerkens and Lipo (2005) demonstrated that the thickness of Rose Springs projectile points from Owens Valley (USA) varied according to the predictions of the random accumulated copying error model. However, the basal width of these projectile points exhibited less variation than expected, suggesting that some non-mutation process (i.e., some form of stabilizing selection such as conformity) was operating on this particular variable.


Principal component analysis

Principal Component Analysis is a technique to extract relevant information from a multivariate dataset and to express this information as a set of few new variables called principal components or dimensions [@cascalheira2018use].

In archaeology, PCA has been often used for~

We applied PCA to our dataset to identify correlations between variables that would represent the existence of patterns in the points, which could possilby be translated into more ~ , in addition to correlation analysis.

```{r}
library(ggbiplot)
library(ggfortify) 

PCA <- prcomp(DF, center = TRUE, scale = TRUE)

# remove stage 1 artefact
xx_pca <- 
df_sitename %>% 
  filter(SPstage.Stage != 1) %>% 
  select(-sitename, -SPstage.Stage) %>% 
  prcomp(., center = TRUE, scale = TRUE)

xx_pca1 <- 
  data.frame(xx_pca$x,
             stage = as.factor( df_sitename$SPstage.Stage[df_sitename$SPstage.Stage !=1]))


## CV with site (n=X)
label_phase <-
   df_sitename %>% 
   group_by(SPstage.Stage) %>% 
   add_tally() %>% 
   mutate(label = as.factor(paste0(SPstage.Stage,'\nN = ',n))) 


count_phase <- length(which(label_phase[9]>0))
count_phase2 <- length(which(label_phase[9] == 2))
count_phase3 <- length(which(label_phase[9] == 3))


###

#plot for both stage
gg_stage <- 
ggplot(xx_pca1 ,
       aes(PC1,
           PC2,
       colour = stage)) +
  geom_point() +
  stat_ellipse()

# phase 2

xx_pca2 <- 
df_sitename %>% 
  filter(SPstage.Stage == 2) %>% 
  select(-sitename, -SPstage.Stage) %>% 
  prcomp(., center = TRUE, scale = TRUE)



# phase 3

xx_pca3 <- 
df_sitename %>% 
  filter(SPstage.Stage == 3) %>% 
  select(-sitename, -SPstage.Stage) %>% 
  prcomp(., center = TRUE, scale = TRUE)



```




We see some differences in artefact shape and size. Now we need to examins the PCA output numbers to describe in more details. 

```{r}

library(factoextra) #http://www.sthda.com/english/articles/31-principal-component-methods-in-r-practical-guide/112-pca-principal-component-analysis-essentials/

# Color variables by the continuous variable
cv_all <- fviz_pca_var(xx_pca, col.var = "contrib",
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")) + ggtitle(paste("All", "(n =", count_phase,")"))

# phase 2
cv_two <- fviz_pca_var(xx_pca2, col.var = "contrib",
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")) + ggtitle(paste("Phase 2", "(n =", count_phase2,")"))
# phase 3
cv_three <- fviz_pca_var(xx_pca3, col.var = "contrib",
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")) + ggtitle(paste("Phase 3", "(n =", count_phase3,")"))



library(patchwork)

cv_all + cv_two + cv_three + plot_layout(ncol=1)

```



Overall, BL and ML have stronger influence than the other attributes and are related to each other but not with the other attributes throughout time. From the phase 2 to the phase 3, the proportion of TL and SL gets bigger and they are more related to SW. 


SYG6 site data and analysis: artifacts from SYG6 site: now it has two layers, but the same phase (2nd)

```{r}

library(stringr)
SYG6_df <- 
  df %>% 
  filter(str_detect(files, "SYG6"))


SYG6_df_sitename <- SYG6_df %>% 
  mutate (files = str_remove_all(files, "_.tps")) %>% 
  separate (files, into = c("sitename", "files"), sep = "_") %>% 
  select (-files)

#list of all attributes without ID
SYG6_DF <- SYG6_df[,1:7]

# make n=element number 
count_SYG6 <- length(which(SYG6_df$SPstage.Stage == 2))


# using corrr

corrr_SYG6_DF <-correlate(SYG6_DF) %>% 
  shave()

cSYG6_DF <- rplot(corrr_SYG6_DF) +
  ggtitle(paste("Correlation SYG6: 2nd Phase", "(n =", count_SYG6,")")) 

  
SYG6_cor <- cor(SYG6_DF)

# CV
map_df(SYG6_DF, cv)

SYG6_PCA <- prcomp(SYG6_cor, center = TRUE, scale = TRUE)

# remove stage 1 artefact
SYG6_xx_pca <- 
SYG6_df_sitename %>% 
  select(-sitename, -SPstage.Stage) %>% 
  prcomp(., center = TRUE, scale = TRUE)

SYG6_xx_pca1 <- 
  data.frame(SYG6_xx_pca$x,
             stage = as.factor( SYG6_df_sitename$SPstage.Stage))

#plot for both stage
SYG6_gg_stage <- 
ggplot(SYG6_xx_pca1 ,
       aes(PC1,
           PC2,
       colour = stage)) +
  geom_point() +
  stat_ellipse()

cv_SYG6 <-fviz_pca_var(SYG6_xx_pca, col.var = "contrib",
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")) +
             ggtitle(paste("SYG6: 2nd Phase", "(n =", count_SYG6,")"))


SYG6_PC1_v <- fviz_contrib(SYG6_xx_pca, choice = "var", axes = 1, top = 7)
SYG6_PC2_v <- fviz_contrib(SYG6_xx_pca, choice = "var", axes = 2, top = 7)

#grid.arrange(SYG6_PC1_v, SYG6_PC2_v, nrow =1) 




```

SYG1 site data and analysis: artifacts from SYG1 site, which include two differnt locations(excavation pits) and the artifacts are from different phase.
SYG1_2 : 3rd phase
SYG1_1: 2nd phase


```{r}
#SYG1_2
library(stringr)
SYG1_2df <- 
  df %>% 
  filter(str_detect(files, "SYG1_2"))

library(dplyr)
SYG1_2df_sitename <- SYG1_2df %>% 
  mutate (files = str_remove_all(files, "_\\d+.tps")) %>% 
  dplyr::rename(sitename = files)
  

#list of all attributes without ID
SYG1_2DF <- SYG1_2df[,1:7]

# make n=element number 
count_SYG1_2 <- length(which(SYG1_2df$SPstage.Stage == 3))



# using corrr

corrr_SYG1_2DF <-correlate(SYG1_2DF) %>% 
  shave()

cSYG1_2DF <- rplot(corrr_SYG1_2DF) +
  ggtitle(paste("Correlation SYG1_2: 3rd Phase", "(n =", count_SYG1_2,")"))




SYG1_2cor <- cor(SYG1_2DF)

# CV
map_df(SYG1_2DF, cv)

SYG1_2PCA <- prcomp(SYG1_2DF, center = TRUE, scale = TRUE)

# remove stage 1 artefact
SYG1_2xx_pca <- 
SYG1_2df_sitename %>% 
  select(-sitename, -SPstage.Stage) %>% 
  prcomp(., center = TRUE, scale = TRUE)

SYG1_2xx_pca1 <- 
  data.frame(SYG1_2xx_pca$x,
             stage = as.factor(SYG1_2df_sitename$SPstage.Stage))

#plot for both stage
SYG1_2gg_stage <- 
ggplot(SYG1_2xx_pca1 ,
       aes(PC1,
           PC2,
       colour = stage)) +
  geom_point() +
  stat_ellipse()

cv_SYG1_2 <- fviz_pca_var(SYG1_2xx_pca, col.var = "contrib",
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")) +
  ggtitle(paste("SYG1_2: 3rd Phase", "(n =", count_SYG1_2,")"))
  

SYG1_2PC1_v <- fviz_contrib(SYG1_2xx_pca, choice = "var", axes = 1, top = 7)
SYG1_2PC2_v <- fviz_contrib(SYG1_2xx_pca, choice = "var", axes = 2, top = 7)

#grid.arrange(SYG1_2PC1_v, SYG1_2PC2_v, nrow =1)



```

```{r}
#SYG1_1
library(stringr)
SYG1_1df <- 
  df %>% 
  filter(str_detect(files, "SYG1_1"))

library(dplyr)
SYG1_1df_sitename <- SYG1_1df %>% 
  mutate (files = str_remove_all(files, "_\\d+.tps")) %>% 
  dplyr::rename(sitename = files)
  

#list of all attributes without ID
SYG1_1DF <- SYG1_1df[,1:7]

# make n=element number 
count_SYG1_1 <- length(which(SYG1_1df$SPstage.Stage == 2))



# using corrr

corrr_SYG1_1DF <-correlate(SYG1_1DF) %>% 
  shave()

cSYG1_1DF <- rplot(corrr_SYG1_1DF) +
  ggtitle(paste("Correlation SYG1_1: 2nd Phase", "(n =", count_SYG1_1,")"))



SYG1_1cor <- cor(SYG1_1DF)

# CV
map_df(SYG1_1DF, cv)

SYG1_1PCA <- prcomp(SYG1_1DF, center = TRUE, scale = TRUE)

# remove stage 1 artefact
SYG1_1xx_pca <- 
  SYG1_1df_sitename %>% 
  select(-sitename, -SPstage.Stage) %>% 
  prcomp(., center = TRUE, scale = TRUE)


SYG1_1xx_pca1 <- 
  data.frame(SYG1_1xx_pca$x,
             stage = as.factor(SYG1_1df_sitename$SPstage.Stage))

#plot for both stage
SYG1_1gg_stage <- 
ggplot(SYG1_1xx_pca1 ,
       aes(PC1,
           PC2,
       colour = stage)) +
  geom_point() +
  stat_ellipse()

cv_SYG1_1 <- fviz_pca_var(SYG1_1xx_pca, col.var = "contrib",
             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07")) +
            ggtitle(paste("SYG1_1: 2nd Phase", "(n =", count_SYG1_1,")"))
             

SYG1_1PC1_v <- fviz_contrib(SYG1_1xx_pca, choice = "var", axes = 1, top = 7)
SYG1_1PC2_v <- fviz_contrib(SYG1_1xx_pca, choice = "var", axes = 2, top = 7)

#grid.arrange(SYG1_1PC1_v, SYG1_1PC2_v, nrow =1)



```



Correlations for each site

```{r}

### three sites at once, using corrr
library(patchwork)
cSYG6_DF + cSYG1_2DF + cSYG1_1DF + plot_layout(ncol=1)

```

In this plot, we can see that the pattern of correlation for SYG6 and SYG1_1 are similar in terms of correlation based on ML/SL/TL with other attributes. Both are part of the phase 2. However, SYG1_2 is younger than the other two and the correlation based on SW become weaker. 

When it comes to the difference between SYG6 and SYG1_1, the correlation between artifacts of SYG1_1 is stronger than SYG6. 


PCA for each site

```{r}
cv_SYG6 + cv_SYG1_1 + cv_SYG1_2 +  plot_layout(ncol=1)
```

Unlike the correlation of the three units, PCA of the three are more various. Especially, SW is related to SL/TL in SYG6, related to ML/BL in SYG1_1, and unrelated to the other attributes in SYG1_2. 

# Results

```{r get-data, eval = FALSE}
# Note the path that we need to use to access our data files when rendering this document
my_data <- read.csv(here::here('analysis', 'data', 'raw_data', 'my_csv_file.csv'))
```

# Discussion

# Conclusion

# Acknowledgements

<!-- The following line inserts a page break when the output is MS Word. For page breaks in PDF, use \newpage on its own line.  -->
##### pagebreak

# References 
<!-- The following line ensures the references appear here for the MS Word or HTML output files, rather than right at the end of the document (this will not work for PDF files):  -->
<div id="refs"></div>

##### pagebreak

### Colophon

This report was generated on `r Sys.time()` using the following computational environment and dependencies: 

```{r colophon, cache = FALSE}
# which R packages and versions?
devtools::session_info()
```

The current Git commit details are:

```{r}
# what commit is this file at? 
git2r::repository(here::here())
```
